_source/_posts/2022-07-21-jhipster-ci-cd.md

```diff
@@ -0,0 +1,865 @@
+
+- Name: DOCKERHUB_PASS
+- Value: your DockerHub password, or better, a DockerHub access token if you have 2FA enabled.
+
+**Note**: For creating a DockerHub access token, sign in to DockerHub, and choose **Account Settings** in the top right user menu. Then, on the left menu, choose **Security**. Click **New Access Token** and set a description for the token, for example, _circleci-store_. Then click **Generate** and copy the new token.
+
+Once a project is set up in CircleCI, a pipeline is triggered each time a commit is pushed to the configured branch. The pipeline in execution appears on the **Dashboard** page. You can also manually trigger the pipeline from the **Dashboard** if you choose the project and branch from the pipeline filters, and then click **Trigger Pipeline**. Before moving on to the next section, manually execute the store pipeline and the gateway pipeline once, to push the first image of each to DockerHub.
+
+{% img blog/jhipster-ci-cd/circleci-job-success.png alt:"CircleCI job success" width:"800" %}{: .center-image }
+
+## Install Spinnaker on Google Kubernetes Engine
+
+Spinnaker supports the following installation environments:
+- Distributed installation on Kubernetes
+- Local installation of Debian packages
+- Local git installation from GitHub
+
+For this example, the distributed alternative was selected. But before the installation, learn about some key concepts in Spinnaker configuration.
+
+### Understand Spinnaker Artifacts and Accounts
+
+A _Spinnaker artifact_ is a named JSON object that refers to an external resource, for example, a docker image or a file stored in GitHub. In a pipeline trigger, you can specify an _expected artifact_, and Spinnaker will compare the incoming artifact from the trigger and bind it to be used by the trigger or another stage in the pipeline.
+Also in Kubernetes, the deployed manifests can be provided statically or as an artifact. In this example, manifests are provided as artifacts stored in GitHub.
+
+For Spinnaker to access and act on resources, like docker registries, cloud providers, and code repositories, different types of accounts must be enabled in Spinnaker configuration, along with the credentials. In addition, the configuration has the optional step of associating Spinnaker with a Kubernetes service account, allowing to restrict permissions over the cluster resources granted to Spinnaker.
+
+### Install Halyard
+
+As described on Spinnaker [docs](https://spinnaker.io/docs/setup/install/), the first step is to install [Halyard](https://spinnaker.io/docs/setup/install/halyard/). For a local install in MacOS:
+
+```shell
+curl -O https://raw.githubusercontent.com/spinnaker/halyard/master/install/macos/InstallHalyard.sh
+sudo bash InstallHalyard.sh
+```
```

Comment:
I see this warning:

```
Java is already installed & at the right version
```

Is there a chance that someone could do this tutorial and use the wrong version of Java?